Description: reduce the rust debuginfo level on selected architectures where
 compiling with debuginfo=2 causes the OOM killer to interrupt the build on
 launchpad builders. Initially this was only on 32 bit architectures, but with
 firefox 63 it started happening frequently on arm64 and ppc64el too, and with
 newer versions it started happening very frequently on s390x too.

--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -2138,9 +2138,9 @@
 
 
 @depends(
-    rustc_opt_level, debug_rust, "--enable-debug-symbols", "--enable-frame-pointers"
+    rustc_opt_level, debug_rust, "--enable-debug-symbols", "--enable-frame-pointers", host
 )
-def rust_compile_flags(opt_level, debug_rust, debug_symbols, frame_pointers):
+def rust_compile_flags(opt_level, debug_rust, debug_symbols, frame_pointers, host):
     # Cargo currently supports only two interesting profiles for building:
     # development and release. Those map (roughly) to --enable-debug and
     # --disable-debug in Gecko, respectively.
@@ -1895,6 +1895,8 @@ def rust_compile_flags(opt_level, debug_
 
     if debug_symbols:
         debug_info = "2"
+        if host.bitness == 32 or host.cpu in ("aarch64", "ppc64", "s390x"):
+            debug_info = "1"
 
     opts = []
 
